File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer = "";
    let players = new Map();
    for(let part of participant){
        players.set(part, (players.get(part) || 0) + 1);
    }
    for(let comp of completion){
       players.set(comp, players.get(comp) - 1);
    }
    for(let [k, v] of players){
        if (v !== 0) {
            answer = k;
            break;
        }
    }

    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    participation = {}
    for person in participant:
        participation[person] = participation.get(person, 0) + 1
    
    for person in completion:
        participation[person] -= 1
    
    for k, v in participation.items():
        if v > 0:
            return k
